src/GovernorCountingFractional.sol

```diff
@@ -41,7 +41,7 @@ abstract contract GovernorCountingFractional is Governor {
      * @dev See {IGovernor-hasVoted}.
      */
     function hasVoted(uint256 proposalId, address account) public view virtual override returns (bool) {
-        return _proposalVotersHasVoted[proposalId][account];
+        return _proposalVotersWeightCast[proposalId][account] > 0;
```

Comment:
Hmm, I think there's a subtle issue here that I'm a little worried about: currently if `hasVoted` returns `true` there is an implicit assumption that they won't vote again and their votes won't change. We're changing that, and I could see it breaking tooling or extension contracts that make that assumption. Not sure what if any action we can take is, though. Hmmm.

Reply:
Interesting point. The [docs](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/bc6de21fe229520b33d13a4d60bbeb1bc5f91bcd/contracts/governance/IGovernor.sol#L185-L189) for `hasVoted` are (understandly) ambiguous on this point:

> Returns whether `account` has cast a vote on `proposalId`.

So we aren't breaking that definition. But to your point people almost certainly have interpreted this as a binary "all or none", since there was no flex voting until now. I think the options are:
1. Do nothing, since this is technically fine. Confirm this is ok and won't break much by searching for some on/off-chain `hasVoted` usage via sourcegraph
2. Add a new `hasFinishedVoting` that returns true if you've voted with all your weight, to ensure there's a method with the original implied semantics
3. Restore the implied semantics of this method and add a `hasPartiallyVoted` method 

I'm personally ok with any of these, perhaps we pull this out into a follow up issue and ask OZ what they'd prefer, to increases chances of getting this upstreamed?

Reply:
Yeah, it's a good point. I'm also not sure what to do about this. Having this return `false` until all weight is cast seems confusing as well -- maybe even more so.

Reply:
Curious what @mds1 thinks of this issue

Reply:
@rsolari (since I know you discussed this feature with @apbendi) I'm curious if you have any take on this question.

TLDR: how should `hasVoted` work if it's possible for people to cast partial votes?

Some options:
1. it could return true so long as some weight has been cast, even if not all weight has been cast
1. it could return false unless all weight has been cast
1. some combination of (1) and then also add another function like `hasFullyVoted` to capture the behavior that `hasVoted` used to have
1. something else… we’re all ears

What would be best from your (Tally's) point of view

Reply:
Tally doesn't use `hasVoted()` right now, so I don't have a strong opinion from Tally's perspective.

I can imagine a frontend using `hasVoted()` to see whether a user is still able to vote. I could also imagine a subgraph/indexer to using it to count up votes cast. 

For a Flexible Voter, both an indexer and a frontend would need to know how many votes are cast so far. We can't answer that with a boolean, so I don't think option 3 really helps. I'd suggest option 1 because it keeps the semantics from `IGovernor`: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/91e8d0ba3c3beb8a1db31310d8599664e48639ef/contracts/governance/IGovernor.sol#L187

I'd suggest adding a helper e.g. `votesCast(uint256 proposalId, address account)` for getting the votes so far. Tally would just use an indexer to avoid RPC calls, but it's nice to give the option to smart contracts and frontends.

Reply:
@davidlaprade mind pulling this out into a separate issue, then we can mark resolved?

Reply:
Awesome, thanks so much for the input @rsolari!

I think I agree that adding a new helper would be the best approach here. Tracking this in https://github.com/ScopeLift/flexible-voting/issues/37